src/plugins/generic/histview/historywindow.cpp

```diff
@@ -276,6 +283,16 @@ void HistoryWindow::fillMonth(QTreeWidgetItem *monthItem)
 	});
 }
 
+void HistoryWindow::clearMonth(QTreeWidgetItem *monthItem) {
+	auto month = monthItem->data(0, Qt::UserRole).toDate();
+
+	auto item = findChild(findChild(ui.dateTreeWidget->invisibleRootItem(), month.year()), month);
+
+	if(item)
+		item->takeChildren();
```

Comment:
Looks like a memory leak, probably qDeleteAll should be called on resulting container